### PR TITLE
add spec for `requiredTransformers` function of `engine.ts`

### PR DESCRIPTION
**Pull request by mayankshukla94**
This PR adds spec for `requiredTransformers` function of `engine.ts`

### CODE REVIEW

**Comment by mayankshukla94** [comment]
@imanpalsingh Please review

File: src/cli/__tests__/engine.spec.ts [hunk 1]
```diff
@@ -104,4 +104,47 @@ describe('Engine', () => {
 
     expect(engine.canExecute(queryToBeExecuted)).toBeTruthy();
   });
+
+  it('determines if function requires transformer', () => {
```
Comment (imanpalsingh) [inline src/cli/__tests__/engine.spec.ts]:
```suggestion
  it('returns correct transformers for a table', () => {
```

File: src/cli/__tests__/engine.spec.ts [hunk 2]
```diff
@@ -104,4 +104,47 @@ describe('Engine', () => {
...
+    };
+
+    const result = engine.requiredTransformers(operators);
+
+    expect(result!.name).toEqual(operators.name);
+    expect(result!.description).toEqual(operators.description);
+    expect(result!.id).toBeUndefined();
+  });
+
+  it('returns null for empty operators when checking for requiredTransformers()', () => {
```
Comment (imanpalsingh) [inline src/cli/__tests__/engine.spec.ts]:
```suggestion
  it('returns null if no transformers are defined, () => {
```

File: src/cli/engine.ts [hunk 3]
```diff
@@ -111,14 +111,14 @@ class Engine {
 
       if (Array.isArray(tableOperations)) {
         const [middleware, transforms] = tableOperations;
-        const requiredTransformers = this.#requiredTransformers({
+        const requiredTransformers = this.requiredTransformers({
```
Comment (imanpalsingh) [inline src/cli/engine.ts]: Can we change name of this variable to something else? it is conflicting with the function name
```suggestion
        const identifiedTransformers = this.requiredTransformers({
```

**Review by imanpalsingh — CHANGES_REQUESTED** [review]

**Review by imanpalsingh — APPROVED** [review]

**Comment by imanpalsingh** [comment]
Thanks 🎉

**Pull request merged**